pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -223,6 +224,66 @@ func (vs *vSphereVMProvider) GetVirtualMachineGuestInfo(
 	return guestInfo, nil
 }
 
+func (vs *vSphereVMProvider) GetVirtualMachineProperties(
+	ctx context.Context,
+	vm *vmopv1.VirtualMachine,
+	propertyPaths []string) (map[string]any, error) {
+
+	vmCtx := pkgctx.VirtualMachineContext{
+		Context: context.WithValue(ctx, vimtypes.ID{}, vs.getOpID(vm, "properties")),
+		Logger:  log.WithValues("vmName", vm.NamespacedName()),
+		VM:      vm,
+	}
+
+	client, err := vs.getVcClient(vmCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	vcVM, err := vs.getVM(vmCtx, client, true)
+	if err != nil {
+		return nil, err
+	}
+
+	propSet := []vimtypes.PropertySpec{{Type: "VirtualMachine"}}
+	if len(propertyPaths) == 0 {
+		propSet[0].All = vimtypes.NewBool(true)
+	} else {
+		propSet[0].PathSet = propertyPaths
+	}
+
+	rep, err := property.DefaultCollector(client.VimClient()).RetrieveProperties(
```

Comment:
@dougm, do you know if it is safe to invoke `RetrieveProperties` from multiple threads in the same session with the `DefaultCollector`?

Reply:
Yes it is safe